README.md

```diff
@@ -4,299 +4,217 @@
+const stringifiedCookies = JSON.stringify(cookies);
 ```
 
-## API
-
-- [CookieConsent()](#cookieconsentoptions-object)
-- [getDialog()](#getdialog)
-- [showDialog()](#showdialog)
-- [hideDialog()](#hidedialog)
-- [isAccepted()](#isacceptedid-string)
-- [getPreferences()](#getpreferences)
-- [on()](#on)
-- [updatePreference()](#updatePreferencecookies-array)
+```html
+<cookie-consent data-cookies="cookies" />;
+```
 
-### CookieConsent(options: object)
+## Options
 
-Will create a new instance.
+To use the options, add them as data attributes to the custom element:
 
 ```js
-const cookieConsent = CookieConsent({
-    cookies: [
-        // ...
-    ]
-});
+<cookie-consent
+    data-title="Cookies & Privacy" // The title of the dialog.
+    data-description="<p>This site makes use of third-party cookies.
+    Read more in our <a href='/privacy-policy'>privacy policy</a>.</p>"  // The description of the dialog.
+    data-saveButtonText="Save preferences" // The save button label.
+    data-cookies=cookies
+/>
+
 ```
 
-To make the instance globally available (for instance to add event listeners elsewhere), add it as a global after the instance has been created:
+All options except `cookies` are optional. They will fall back to the defaults, which are listed here:
 
 ```js
-const cookieConsent = CookieConsent();
-
-window.CookieConsent = cookieConsent;
+export const DEFAULTS = {
+    type: "checkbox",
+    prefix: "cookie-consent",
+    append: true,
+    appendDelay: 500,
+    acceptAllButton: false,
+    labels: {
+        title: "Cookies & Privacy",
+        description:
+            '<p>This site makes use of third-party cookies. Read more in our <a href="/privacy-policy">privacy policy</a>.</p>',
+        button: {
+            default: "Save preferences",
+            acceptAll: "Accept all",
+        },
+        aria: {
+            button: "Confirm cookie settings",
+            tabList: "List with cookie types",
+            tabToggle: "Toggle cookie tab",
+        },
+    },
+};
 ```
 
-### getDialog()
-
-Will fetch the dialog element, for example to append it at a custom DOM position.
+## API
 
-```js
-document.body.insertBefore(cookieConsent.getDialog(), document.body.firstElementChild);
-```
+-   [show()](#showdialog)
+-   [hide()](#hidedialog)
 
-### showDialog()
+### show()
 
 Will show the dialog element, for example to show it when triggered to change settings.
 
 ```js
-el.addEventListener('click', e => {
-  e.preventDefault();
-  cookieConsent.showDialog();
+button.addEventListener("click", (e) => {
+    e.preventDefault();
+    cookieConsent.show();
 });
 ```
 
-### hideDialog()
+### hide()
 
 Will hide the dialog element.
 
 ```js
-el.addEventListener('click', e => {
-  e.preventDefault();
-  cookieConsent.hideDialog();
+el.addEventListener("click", (e) => {
+    e.preventDefault();
+    cookieConsent.hide();
 });
 ```
 
-### isAccepted(id: string)
```

Comment:
Ah yeah, I didn't see. Everything has been re-added except for `isAccepted` this method is not being used anymore as this can be easily checked in the frontend right now after cookies have been updated.